### DISCUSSION TITLE
TextBlock.SelectAll not support in UNO MacCatalyst

### DISCUSSION THREAD

**Discussion by rruprai1** [unoplatform/uno#17811] [General]
Is there a PR or bug to indicate that TextBlock.SelectAll is not supported in UNO for MacCatalyst (using Uno skd 5.3.31)

Or is that to be expected? i'm not sure if I need to cast to something more generic or specific for targetting macos.

**Comment by MartinZikmund** [comment]
The API should be supported, but we will verify on our end.

**Reply by agneszitte** [reply]
> The API should be supported, but we will verify on our end.

Hi @rruprai1,
@MartinZikmund thought of `TextBox` instead of `TextBlock` when writing this. Currently, `TextBlock.SelectAll` is only available for Skia. We have an open issue https://github.com/unoplatform/uno/issues/16537 to make it available for WASM. I'll verify with @MartinZikmund whether we should adjust this issue or open a new feature request to support it on other platforms.

**Reply by rruprai1** [reply]
Cool. thanks.

**Reply by MartinZikmund** [reply]
For non-Skia and non-WASM targets, implementing this API will be particularly complex, as we perform the text rendering in a custom way, which does not provide the "native" text selection affordances.
